query-planner-js/src/buildPlan.ts

```diff
@@ -810,7 +813,21 @@ class FetchGroup {
     // key for that. Having it here saves us from re-computing it more than once.
     readonly subgraphAndMergeAtKey?: string,
     private cachedCost?: number,
+    // Cache used to save unecessary recomputation of the `isUseless` method.
+    private isKnownUseful: boolean = false,
```

Comment:
Wouldn't it be better to cache the status either way? i.e. the type should be `boolean | undefined` and you can just return the value if it's not undefined and run isUseful() otherwise? Obviously you should change the variable name if you do that.

Reply:
I don't think it'd be better. First, as I mentioned in one of the comment of the patch, when a group is shown to be useless, it is removed right away, so it doesn't really make matter whether that information is cached or not But adding to that, to properly cache the status either way, we'd have to add invalidation for that other way: that, changes to the group `selection` would have to invalid this cache, when currently only change to the `inputs` (which happens to be less frequent in practice) does so. So implementing the other would make things possibly less efficient overall.